src/main/java/uk/gov/justice/digital/hmpps/whereabouts/services/AppointmentService.kt

```diff
@@ -78,18 +78,19 @@ class AppointmentService(
     val mainAppointmentDetails: PrisonAppointment = prisonApiService.getPrisonAppointment(appointmentId)
       ?: throw EntityNotFoundException("Appointment $appointmentId does not exist")
 
-    val offenderNo = prisonApiService.getOffenderNoFromBookingId(mainAppointmentDetails.bookingId)
+    val offenderNo = try {
+      prisonApiService.getOffenderNoFromBookingId(mainAppointmentDetails.bookingId)
+    } catch (e: Exception) {
+      null
```

Comment:
We need a good reason to ignore exceptions - is it the case here that this is just extra data that shouldn't stop the rest of the data from being returned?

Reply:
I did some testing and for some offenders I was getting a 404. I'll do some more testing and maybe remove or add some logging.